ddclient.in

```diff
@@ -2155,6 +2168,171 @@ sub geturl {
     return $reply;
 }
 
+######################################################################
+## curl_cmd() function to execute system curl command
+######################################################################
+sub curl_cmd {
+    my @params = @_;
+    my $tmpfile;
+    my $tfh;
+    my $system_curl = subst_var('@syscurlcmd@', '/usr/bin/curl');
+    my %curl_codes = (  ## Subset of error codes from https://curl.haxx.se/docs/manpage.html
+         2 => "Failed to initialize. (Most likely a bug in ddclient, please open issue at https://github.com/ddclient/ddclient)",
+         3 => "URL malformed. The syntax was not correct",
+         5 => "Couldn't resolve proxy. The given proxy host could not be resolved.",
+         6 => "Couldn't resolve host. The given remote host was not resolved.",
+         7 => "Failed to connect to host.",
+        22 => "HTTP page not retrieved. The requested url was not found or returned another error.",
+        28 => "Operation timeout. The specified time-out period was reached according to the conditions.",
+        35 => "SSL connect error. The SSL handshaking failed.",
+        47 => "Too many redirects. When following redirects, curl hit the maximum amount.",
+        51 => "The peer's SSL certificate or SSH MD5 fingerprint was not OK.",
+        58 => "Problem with the local certificate.",
+        60 => "Peer certificate cannot be authenticated with known CA certificates.",
+        67 => "The user name, password, or similar was not accepted and curl failed to log in.",
+        77 => "Problem with reading the SSL CA cert (path? access rights?).",
+        78 => "The resource referenced in the URL does not exist.",
+       127 => "You requested network access with curl but $system_curl was not found",
+    );
+
+    return '' if (scalar(@params) == 0); ## no parameters provided
+
+    # debug("CURL: %s\r\n",join("\r\n      ",@params)); ## Uncomment only if required as may expose credentials
+
+    ## Use system File::Temp to generate filename...
+    use File::Temp;
+    $tfh = File::Temp->new( TMPDIR => 1,  ## in system temp dir
+                            UNLINK => 0,  ## we will delete it ourselves
+                            TEMPLATE => 'ddclient_XXXXXXXXXX');
+    $tmpfile = $tfh->filename;
+
+    chmod(0600,$tmpfile);
+    debug("CURL Tempfile: %s",$tmpfile);
+    local $\ = "\n";  ## Terminate the file,
+    local $, = "\n";  ## and each parameter, with a newline.
+    print($tfh @params);
+    close($tfh);
+
+    ## Make sure to delete temp file on Ctrl-C or SIGTERM
+    local $SIG{'HUP'} = sub { unlink($tmpfile); $caught_hup = 1; exit 1 if !$daemon; };
+    local $SIG{'INT'} = sub { unlink($tmpfile); $caught_term = 1; exit 2 if !$daemon; };
+    local $SIG{'TERM'} = sub { unlink($tmpfile); $caught_int = 1; exit 15 if !$daemon; };
```

Comment:
Delete this signal handling. It only makes a difference if curl is running outside the main loop and a signal arrives, which never happens in normal operation. Even if it did, the readability and maintenance cost is much worse than the occasional lingering temp file.

Reply:
Done